### ISSUE TITLE
letsencrypt-auto doesn't fail fast

### ISSUE THREAD

**Issue by kuba** [certbot/certbot#1786]
In case of `apt-get install` errors, `letsencrypt-auto` happily continues with the bootstrap:
```
Bootstrapping dependencies for Debian-based OSes...
Hit http://ppa.launchpad.net trusty InRelease
Ign http://overviewer.org ./ InRelease
Hit http://overviewer.org ./ Release.gpg
Hit http://overviewer.org ./ Release
Hit http://ppa.launchpad.net trusty/main i386 Packages
Hit http://overviewer.org ./ Packages
Hit http://ppa.launchpad.net trusty/main Translation-en
Ign http://overviewer.org ./ Translation-en_US
Ign http://overviewer.org ./ Translation-en
Hit https://packages.gitlab.com trusty InRelease
Hit https://packages.gitlab.com trusty/main Sources
Hit https://packages.gitlab.com trusty/main i386 Packages
Get:1 https://packages.gitlab.com trusty/main Translation-en_US
Ign https://packages.gitlab.com trusty/main Translation-en_US
Ign https://packages.gitlab.com trusty/main Translation-en
Reading package lists... Done
E: No packages found
Reading package lists... Done
Building dependency tree
Reading state information... Done
Package dialog is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package python-virtualenv is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'python-virtualenv' has no installation candidate
E: Package 'dialog' has no installation candidate
E: Unable to locate package libaugeas0
E: Unable to locate package libssl-dev
E: Unable to locate package libffi-dev
Creating virtual environment...
Running virtualenv with interpreter /usr/bin/python2.7
New python executable in /root/.local/share/letsencrypt/bin/python2.7
Also creating executable in /root/.local/share/letsencrypt/bin/python
Installing setuptools, pip, wheel...done.
Updating letsencrypt and virtual environment dependencies...
Collecting setuptools
  Using cached setuptools-18.7.1-py2.py3-none-any.whl
Installing collected packages: setuptools
  Found existing installation: setuptools 18.2
    Uninstalling setuptools-18.2:
      Successfully uninstalled setuptools-18.2
Successfully installed setuptools-18.7.1
/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:90: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Requirement already up-to-date: pip in /root/.local/share/letsencrypt/lib/python2.7/site-packages
Collecting letsencrypt
  Using cached letsencrypt-0.1.0-py2-none-any.whl
Collecting letsencrypt-apache
  Using cached letsencrypt_apache-0.1.0-py2-none-any.whl
Collecting ConfigArgParse from git+https://github.com/kuba/ConfigArgParse.git@python2.6-0.9.3#egg=ConfigArgParse (from -r ./py26reqs.txt (line 1))
  Cloning https://github.com/kuba/ConfigArgParse.git (to python2.6-0.9.3) to /tmp/pip-build-3R_dqM/ConfigArgParse
Collecting acme==0.1.0 (from letsencrypt)
  Using cached acme-0.1.0-py2-none-any.whl
Requirement already up-to-date: setuptools in /root/.local/share/letsencrypt/lib/python2.7/site-packages (from letsencrypt)
Collecting python2-pythondialog>=3.2.2rc1 (from letsencrypt)
Collecting PyOpenSSL (from letsencrypt)
  Using cached pyOpenSSL-0.15.1-py2.py3-none-any.whl
Collecting requests (from letsencrypt)
  Using cached requests-2.8.1-py2.py3-none-any.whl
Collecting parsedatetime (from letsencrypt)
  Using cached parsedatetime-1.5-py2-none-any.whl
Collecting configobj (from letsencrypt)
Collecting pytz (from letsencrypt)
  Using cached pytz-2015.7-py2.py3-none-any.whl
Collecting psutil>=2.1.0 (from letsencrypt)
Collecting six (from letsencrypt)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting cryptography>=0.7 (from letsencrypt)
  Using cached cryptography-1.1.1.tar.gz
Collecting zope.interface (from letsencrypt)
Collecting zope.component (from letsencrypt)
Collecting mock (from letsencrypt)
  Using cached mock-1.3.0-py2.py3-none-any.whl
Collecting pyrfc3339 (from letsencrypt)
  Using cached pyRFC3339-1.0-py2.py3-none-any.whl
Collecting python-augeas (from letsencrypt-apache)
Collecting pyasn1 (from acme==0.1.0->letsencrypt)
  Using cached pyasn1-0.1.9-py2.py3-none-any.whl
Collecting ndg-httpsclient (from acme==0.1.0->letsencrypt)
Collecting werkzeug (from acme==0.1.0->letsencrypt)
  Using cached Werkzeug-0.11.2-py2.py3-none-any.whl
Collecting idna>=2.0 (from cryptography>=0.7->letsencrypt)
  Using cached idna-2.0-py2.py3-none-any.whl
Collecting enum34 (from cryptography>=0.7->letsencrypt)
Collecting ipaddress (from cryptography>=0.7->letsencrypt)
  Using cached ipaddress-1.0.15-py27-none-any.whl
Collecting cffi>=1.1.0 (from cryptography>=0.7->letsencrypt)
  Using cached cffi-1.3.1.tar.gz
Collecting zope.event (from zope.component->letsencrypt)
Collecting funcsigs (from mock->letsencrypt)
  Using cached funcsigs-0.4-py2.py3-none-any.whl
Collecting pbr>=0.11 (from mock->letsencrypt)
  Using cached pbr-1.8.1-py2.py3-none-any.whl
Collecting pycparser (from cffi>=1.1.0->cryptography>=0.7->letsencrypt)
Building wheels for collected packages: cryptography, cffi
  Running setup.py bdist_wheel for cryptography
  Complete output from command /root/.local/share/letsencrypt/bin/python2.7 -c "import setuptools;__file__='/tmp/pip-build-3R_dqM/cryptography/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmpWCbyvwpip-wheel-:
  c/_cffi_backend.c:13:17: fatal error: ffi.h: No such file or directory
   #include <ffi.h>
                   ^
  compilation terminated.
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-build-3R_dqM/cryptography/setup.py", line 318, in <module>
      **keywords_with_side_effects(sys.argv)
    File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
      _setup_distribution = dist = klass(attrs)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 268, in __init__
      self.fetch_build_eggs(attrs['setup_requires'])
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 313, in fetch_build_eggs
      replace_conflicting=True,
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 846, in resolve
      dist = best[req.key] = env.best_match(req, ws, installer)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1091, in best_match
      return self.obtain(req, installer)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1103, in obtain
      return installer(requirement)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/dist.py", line 380, in fetch_build_egg
      return cmd.easy_install(req)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 639, in easy_install
      return self.install_item(spec, dist.location, tmpdir, deps)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 669, in install_item
      dists = self.install_eggs(spec, download, tmpdir)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 853, in install_eggs
      return self.build_and_install(setup_script, setup_base)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1081, in build_and_install
      self.run_setup(setup_script, setup_base, args)
    File "/root/.local/share/letsencrypt/local/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1069, in run_setup
      raise DistutilsError("Setup script exited with %s" % (v.args[0],))
  distutils.errors.DistutilsError: Setup script exited with error: command 'i686-linux-gnu-gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for cryptography
  Running setup.py bdist_wheel for cffi
  Complete output from command /root/.local/share/letsencrypt/bin/python2.7 -c "import setuptools;__file__='/tmp/pip-build-3R_dqM/cffi/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmptMRAhApip-wheel-:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-i686-2.7
  creating build/lib.linux-i686-2.7/cffi
  copying cffi/gc_weakref.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/cparser.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/cffi_opcode.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/api.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/model.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/commontypes.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/vengine_cpy.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/backend_ctypes.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/verifier.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/lock.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/vengine_gen.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/setuptools_ext.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/__init__.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/ffiplatform.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/recompiler.py -> build/lib.linux-i686-2.7/cffi
  copying cffi/_cffi_include.h -> build/lib.linux-i686-2.7/cffi
  copying cffi/parse_c_type.h -> build/lib.linux-i686-2.7/cffi
  running build_ext
  building '_cffi_backend' extension
  creating build/temp.linux-i686-2.7
  creating build/temp.linux-i686-2.7/c
  i686-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DUSE__THREAD -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python2.7 -c c/_cffi_backend.c -o build/temp.linux-i686-2.7/c/_cffi_backend.o
  c/_cffi_backend.c:13:17: fatal error: ffi.h: No such file or directory
   #include <ffi.h>
                   ^
  compilation terminated.
  error: command 'i686-linux-gnu-gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for cffi
Failed to build cryptography cffi
Installing collected packages: pytz, requests, pyasn1, idna, six, enum34, ipaddress, pycparser, cffi, cryptography, PyOpenSSL, ndg-httpsclient, werkzeug, funcsigs, pbr, mock, pyrfc3339, acme, python2-pythondialog, parsedatetime, configobj, psutil, zope.interface, zope.event, zope.component, ConfigArgParse, letsencrypt, python-augeas, letsencrypt-apache
  Running setup.py install for cffi
    Complete output from command /root/.local/share/letsencrypt/bin/python2.7 -c "import setuptools, tokenize;__file__='/tmp/pip-build-3R_dqM/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Xv7dzT-record/install-record.txt --single-version-externally-managed --compile --install-headers /root/.local/share/letsencrypt/include/site/python2.7/cffi:
    running install
    running build
    running build_py
    running build_ext
    building '_cffi_backend' extension
    i686-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DUSE__THREAD -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python2.7 -c c/_cffi_backend.c -o build/temp.linux-i686-2.7/c/_cffi_backend.o
    c/_cffi_backend.c:13:17: fatal error: ffi.h: No such file or directory
     #include <ffi.h>
                     ^
    compilation terminated.
    error: command 'i686-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/root/.local/share/letsencrypt/bin/python2.7 -c "import setuptools, tokenize;__file__='/tmp/pip-build-3R_dqM/cffi/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Xv7dzT-record/install-record.txt --single-version-externally-managed --compile --install-headers /root/.local/share/letsencrypt/include/site/python2.7/cffi" failed with error code 1 in /tmp/pip-build-3R_dqM/cffi
```

This was experienced by iamJannik on IRC.

**Comment by doublehops** [comment]
I'm also getting this error when I try `./letsencypt-auto --help` on Debian 8 Jessie.

**Comment by bmw** [comment]
This was resolved in #3111.
